### PR TITLE
Initial work on the backend for auth

### PR DESCRIPTION
**Pull request by LogicX1**
rel #42
Initial setup to check and authenticate a user based on the token,did not add the routes yet

### CODE REVIEW

File: server/app/routes/index.js [hunk 1]
```diff
@@ -25,10 +41,18 @@ app.use(
   cors(corsOptions),
 );
 
+app.use(jwtCheck);
```
Comment (Karyum) [inline server/app/routes/index.js]: Let's no have the middlewares, the server setup and the router in the same file, it's just going to be a mess in the future.

i would add a file called `app.js` outside the routes folder, and in this file `routes/index.js` i would use the `express.Router()` method to make the router modular

Comment (Karyum) [reply server/app/routes/index.js]: [click this link to know more about `express.Router()`](https://expressjs.com/en/guide/routing.html#express-router)

File: server/app/routes/index.js [hunk 2]
```diff
@@ -3,7 +3,23 @@ const morgan = require('morgan');
...
+      secret: jwks.expressJwtSecret({
+          cache: true,
+          rateLimit: true,
+          jwksRequestsPerMinute: 5,
+          jwksUri: `https://${process.env.AUTH0_DOMAIN}/.well-known/jwks.json`
+    }),
+    audience: `https://${process.env.AUTH0_AUDIENCE}/`,
+    issuer: `https://${process.env.AUTH0_DOMAIN}/`,
+    algorithms: ['RS256']
+});
```
[the remark this answers is not in the document]
Comment (LogicX1) [reply server/app/routes/index.js]: if i understand correctly this is jsonwebtoken(jwt-express),  coupled up with the auth0 service the usage is getting weird, they are pulling jwks from the jwksUri that is specific for our app

File: server/app/routes/index.js (hunk 1 above)
Comment (LogicX1) [reply server/app/routes/index.js]: I agree I'll fix that soon, so ill split all the routes to the router, then all middlewares and set up to another on the app.js

**Comment by Karyum** [comment]
@LogicX1 @aysam199 is doing is also splitting the files, we have to coordinate together on this

**Pull request merged**